CAN-11

Lukyix opened the pull request:
Create Navbar component

Lukyix committed:
Add initial Navbar component files

Created the Navbar component structure with empty TypeScript, SCSS, and test files to set up the foundation for future development.

Lukyix committed:
Replace ExampleComponent with Navbar in App

Updated App.tsx to render the Navbar component instead of ExampleComponent. This change reflects a shift in the main UI component displayed in the application.

Lukyix committed:
Add hamburger menu SCSS styles

Introduces SCSS styles for a responsive hamburger menu button, including open state animations and hover effects. The hamburger is hidden by default and shown on small desktop screens and below.

Lukyix committed:
Add Hamburger menu button component

Introduces a reusable Hamburger button component for the navbar with open/close state and optional toggle callback. Includes basic structure and styling hook.

Lukyix committed:
Add responsive styles for navbar component

Expanded navbar.scss with styles for desktop and mobile layouts, including flexbox structure, color variables, and responsive behavior for navigation links, search bar, and hamburger menu. Introduces media queries for small desktop breakpoints to improve mobile usability.

Lukyix committed:
Add responsive Navbar component

Introduces a new Navbar component with desktop and mobile navigation, including a hamburger menu for toggling the mobile menu. The component uses React state to manage menu visibility and includes placeholder links and a search bar.

Lukyix committed:
Add Storybook stories for Navbar component

Introduces a new Storybook story file for the Navbar component, enabling visual testing and documentation under the 'CantusCatholici/Navbar' category.

SamuelSlavik committed:
Merge remote-tracking branch 'origin/CAN-11' into CAN-11

# Conflicts:
#	src/App.tsx

SamuelSlavik commented on src/components/navbar/hamburger.scss:
```diff
@@ -0,0 +1,49 @@
+    }
+}
+
+@media (max-width: $break-point-small-desktop) {
```
This can be inside the .hamburger {}, you can put there the whole @media structure, and you can remove the .hamburger {} and leave just the display: flex

SamuelSlavik committed:
Fix storybook

SamuelSlavik commented on src/components/navbar/Hamburger.tsx:
```diff
@@ -0,0 +1,28 @@
+import { useState } from 'react';
+import './hamburger.scss';
+
+interface HamburgerProps {
```
Rewrite this to "type HamburgerProps" instead of the interface. This is exact type, interface is slightly different structure, that works the same, but has slightly different meaning and some advanced logic that we dont have to use here.

SamuelSlavik commented on src/components/navbar/hamburger.scss:
```diff
@@ -0,0 +1,49 @@
+    cursor: pointer;
+    padding: 0;
+    
+    &__line {
```
For better readability, this should be written as 

.hamburger {
  &__ {
    &line {}
  }
}

You can follow this pattern throughout other instances

SamuelSlavik commented on src/components/navbar/navbar.scss:
```diff
@@ -0,0 +1,121 @@
+}
+
+
+@media (max-width: $break-point-small-desktop) {
```
Can also be broke down and moved inside the selectors

SamuelSlavik commented on src/components/navbar/Hamburger.tsx:
```diff
@@ -0,0 +1,28 @@
+export function Hamburger({ onToggle }: HamburgerProps) {
+    const [isOpen, setIsOpen] = useState(false);
+
+    const handleClick = () => {
```
Probably wont be necesery after moving isOpen to the props

SamuelSlavik commented on src/components/navbar/navbar.tsx:
```diff
@@ -0,0 +1,68 @@
+                    </Link>
+                </div>
+                <div className="hamburger-container">
+                    <Hamburger onToggle={handleMenuToggle} />
```
Add prop isOpen or isToggled

SamuelSlavik commented on src/components/navbar/Hamburger.tsx:
```diff
@@ -0,0 +1,28 @@
+}
+
+export function Hamburger({ onToggle }: HamburgerProps) {
+    const [isOpen, setIsOpen] = useState(false);
```
Set as a prop, so navigation can provide the only source of truth if the menu should be opened or not. Then there is no need set the state on two places, which can be also slightly complicated in this case

Lukyix committed:
Refactor Navbar menu toggle and remove search bar

Simplified the handleMenuToggle function to toggle the mobile menu state instead of setting it directly. Removed the unused search bar from the Navbar. Updated Hamburger component usage to pass isToggled prop.

Lukyix committed:
Refactor navbar responsive styles for clarity

Moved media queries for .search-bar, .nav-links, and .hamburger-container inside their respective class selectors within nav. Removed redundant global media query at the end of the file for improved maintainability and readability.

Lukyix committed:
Refactor Hamburger to controlled component

Changed Hamburger to accept isToggled and onToggle props, removing internal state. This allows parent components to control the open state, improving flexibility and state management.

Lukyix committed:
Refactor hamburger SCSS for improved structure and media query

Moved the media query for small desktop breakpoint inside the .hamburger selector and adjusted the BEM structure for the line element. Cleaned up indentation and removed redundant global media query.

Lukyix committed:
Replace ExampleComponent with Navbar in App

Swapped out the ExampleComponent for the Navbar component in the App, updating the main UI structure.

Lukyix committed:
Refactor navbar to use BEM utility and update styles

Replaced hardcoded class names in Navbar and Hamburger components with the useBem hook for BEM-style class generation. Updated SCSS files to match new BEM structure and fixed indentation and selector nesting for consistency. This improves maintainability and enforces a consistent naming convention.

SamuelSlavik committed:
Merge branch 'dev' into CAN-11

# Conflicts:
#	src/App.tsx

SamuelSlavik committed:
Refactor classnames

SamuelSlavik committed:
Refactor styles

SamuelSlavik committed:
Translate aria-label
